torch/csrc/jit/passes/lower_tuples.cpp

```diff
@@ -40,6 +48,17 @@ static void VisitNode(Node* n, Node* insert_point) {
     }
     return;
   }
+
+  // undo TupleUnpack(TupleIndex)
+  if (n->kind() == prim::TupleIndex) {
+    auto construct = n->inputs().at(0)->node();
+    auto idx = indexVal(n->inputs().at(1));
+    JIT_ASSERTM(construct->kind() == prim::TupleConstruct, "tuple index not matched to tuple construct");
```

Comment:
This is not guaranteed to be the case, which means it shouldn't be an assert (but at most a check). Also, this code is usually meant to lower everything it can without throwing an error, which will be raised by `EnsureNoTuples` later.

Reply:
You're right that idx should be after the assertion. But i think the assertion should remain to maintain convention with TupleUnpack